pkg/referrerstore/oras/oras.go

```diff
@@ -287,8 +287,9 @@ func (store *orasStore) createRepository(ctx context.Context, targetRef common.R
 		return nil, time.Now(), fmt.Errorf("auth provider not properly enabled")
 	}
 
-	if cacheEntry, ok := store.authCache[targetRef.Original]; ok {
+	if entry, ok := store.authCache.Load(targetRef.Original); ok {
```

Comment:
discussed offline: yes ideally we should move to a model where we entirely rely on the oras auth cache internal implementation rather than have our cache we maintain. I think we should queue this up for a future refactor of ORAS store.

Reply:
Some open questions if we switch to oras build in auth cache: 1. we currently use an expiry time to know if creds are valid or not still. how is that handled with oras? 2. If auth creds are bad (failed request), does oras evict the creds from the cache? 3. I think we have to build some sort of retry mechanism so that if the first requrests fails due to some auth issue, it will retry the same operation but first call `Provide` to fetch fresh creds.

Reply:
Oras doesn't support expiry for now, so we cannot move to oras built-in cache entirely. But for the authCache map, I feel we can map it to `authConfig` object instead of a `respository` object. And since it's not the focus of this PR, we can ignore it for now.